nucypher/characters/lawful.py

```diff
@@ -223,7 +221,7 @@ def generate_kfrags(self,
         bob_encrypting_key = bob.public_keys(DecryptingPower)
         delegating_power = self._crypto_power.power_ups(DelegatingPower)
         return delegating_power.generate_kfrags(bob_pubkey_enc=bob_encrypting_key,
-                                                signer=self.stamp,
+                                                signer=self.stamp.as_umbral_signer(),
```

Comment:
The signer has to go inside Umbral, so it has to be an Umbral object. We have an arbitrary Python callable inside Rust code

Reply:
To clarify my question:

Why pass the umbral signer instead of the stamp and then use its `__call__` like we were doing?  What's the advantage of the signer object that the stamp itself can't satisfy?

Reply:
It's called inside `generate_kfrags()`, which is a Rust function. There is no Python available at that level. The signer object is a wrapped Rust object, so it can go through.